### DISCUSSION TITLE
Using truffle debugger in hardhat project

### DISCUSSION THREAD

**Discussion by tomasz90** [ConsenSys-archive/truffle#5387] [General Q&A] [answered]
Hi! 
Unfortunately I ma working on hardhat project, but debugging using console.log is painful..
I am missing truffle debugger, is it possible to use it inside hardhat project?

**Comment by apdan510** [comment]
@tomasz90 

Is your project set up as a hardhat project or Truffle project? Any chance you can share a public repo showing the layout of your directory?

Thank you,
Alex | Truffle Support

**Reply by emilyJLin95** [reply]
I just want to chime in here
@tomasz90 the truffle debugger is unavailable for hardhat currently! 

Here at Truffle we are all about collaboration, so Hard Hat compatibility for our VS Code Extension debugger is on our roadmap, but I can't give a clear date on when.

As for programatic debugging through the CLI or in tests, I don't think we have that planned
@kevinbluer can you verify that that's correct?

**Reply by kevinbluer** [reply]
Thanks for looping me in @emilyJLin95! In theory there's nothing stopping the debugger working with a Hardhat project, although there are a few caveats: first up you'd need to generate the Truffle artifacts (as the debugger currently relies on these to work)...secondly, you'd need to deploy to a locally running Ganache instance.

As @emilyJLin95 mentioned, it might be easier to go the route of using the [Truffle for VS Code](https://marketplace.visualstudio.com/items?itemName=trufflesuite-csi.truffle-vscode) extension which has the benefit of the built-in IDE debugger ([guide to using it here](https://trufflesuite.com/blog/build-on-web3-with-truffle-vs-code-extension/)). We're also actively working on compatibility, which should be available in the not too distant future.

Lastly, as mentioned if you do have a repo you can share, we can take a look at give some additional guidance on getting things setup 👍

**Reply by tomasz90** [reply]
My project is set up as Hardhat. And the other thing is that I have tests written in typescript using typechain. This is another issues as I see. I am getting error cannot use import statement outside a module ;(

**Comment by gnidan** [comment]
This discussion inspired me to look into it, and I think this might be easier than we previously thought! It's still a hack, but hopefully I'll be able to push this in the direction of being mergeable in a few days... stay tuned :)

![hardhat-truffle-debug](https://user-images.githubusercontent.com/151065/183332142-e7925655-d572-4b90-ac4a-caa14334331a.gif)

**Comment by gnidan** [comment, marked as answer]
We just released experimental support for `truffle debug` inside Hardhat projects!

See release notes for [Truffle v5.5.27](https://github.com/trufflesuite/truffle/releases/tag/v5.5.27) for more.

**Reply by gnidan** [reply]
Please let us know if you run into any trouble! Thanks again for starting the discussion here.
